tst/EnergyPlus/unit/HeatBalanceAirManager.unit.cc

```diff
@@ -755,32 +798,12 @@ TEST_F(EnergyPlusFixture, HeatBalanceAirManager_GetMixingAndCrossMixing)
     state->dataIPShortCut->rNumericArgs.dimension(MaxNumeric, 0.0);
     state->dataIPShortCut->lNumericFieldBlanks.dimension(MaxNumeric, false);
 
+    // Need to do this before ManageSimulation to get the space heat balance input
     bool ErrorsFound = false;
-    HeatBalanceManager::GetHeatBalanceInput(*state);
-    std::string const error_string = delimited_string(
-        {"   ** Warning ** GetSurfaceData: Entered Space Floor Area(s) differ more than 5% from calculated Space Floor Area(s).",
-         "   **   ~~~   ** ...use Output:Diagnostics,DisplayExtraWarnings; to show more details on individual Spaces.",
-         "   ** Warning ** CalculateZoneVolume: 1 zone is not fully enclosed. For more details use:  Output:Diagnostics,DisplayExtrawarnings; ",
-         "   ** Warning ** CalcApproximateViewFactors: Zero area for all other zone surfaces.",
-         "   **   ~~~   ** Happens for Surface=\"DUMMY SPACE 1A FLOOR\" in Zone=ZONE 1",
-         "   ** Warning ** CalcApproximateViewFactors: Zero area for all other zone surfaces.",
-         "   **   ~~~   ** Happens for Surface=\"DUMMY SPACE 1B FLOOR\" in Zone=ZONE 1",
-         "   ** Warning ** Surfaces in Zone/Enclosure=\"ZONE 1\" do not define an enclosure.",
-         "   **   ~~~   ** Number of surfaces <= 3, view factors are set to force reciprocity but may not fulfill completeness.",
-         "   **   ~~~   ** Reciprocity means that radiant exchange between two surfaces will match and not lead to an energy loss.",
-         "   **   ~~~   ** Completeness means that all of the view factors between a surface and the other surfaces in a zone add up to unity.",
-         "   **   ~~~   ** So, when there are three or less surfaces in a zone, EnergyPlus will make sure there are no losses of energy but",
-         "   **   ~~~   ** it will not exchange the full amount of radiation with the rest of the zone as it would if there was a completed "
-         "enclosure."});
-
-    compare_err_stream(error_string, true);
+    HeatBalanceManager::GetProjectControlData(*state, ErrorsFound);
     EXPECT_FALSE(ErrorsFound);
 
-    state->dataHeatBalFanSys->ZoneReOrder.allocate(state->dataGlobal->NumOfZones);
-    ErrorsFound = false;
-    HeatBalanceAirManager::GetSimpleAirModelInputs(*state, ErrorsFound);
-    compare_err_stream("", true);
-    EXPECT_FALSE(ErrorsFound);
+    SimulationManager::ManageSimulation(*state);
```

Comment:
Extended this unit test to do an actual simulation so that it tests #10670 

Running the revised unit test with develop gives a vector error:

![image](https://github.com/user-attachments/assets/8f865b93-8141-44a6-ba30-4af109077270)

Because `fromSpaceIndex=0` here:

![image](https://github.com/user-attachments/assets/a4fea8c3-351f-4ee9-8029-03c505e9075f)